src/workflows-app/utils/submission-utils.ts

```diff
@@ -89,37 +107,54 @@ export const inputTypeParamDefaults = {
   object_builder: { fields: [] },
 };
 
-export const isInputOptional = (ioType) => _.get('type', ioType) === 'optional';
+export const isInputOptional = (ioType: InputType): ioType is OptionalInputType => ioType.type === 'optional';
 
-export const inputTypeStyle = (iotype) => {
+export const inputTypeStyle = (iotype: InputType) => {
   if (isInputOptional(iotype)) {
     return { fontStyle: 'italic' };
   }
   return {};
 };
 
-export const renderTypeText = (iotype) => {
-  if (_.has('primitive_type', iotype)) {
+export const renderTypeText = (iotype: InputType): string => {
+  if (iotype.type === 'primitive') {
```

Comment:
This is a cool refactor, presumably made possible (or necessary) because we're switching the file over to ts? Where is this `type` function defined? I couldn't find it over in `submission-models` but did I just miss it? Or is it automatically inferred somehow?